phpstan-baseline.neon.dist

```diff
@@ -1,12 +1,12 @@
 parameters:
 	ignoreErrors:
 		-
-			message: "#^Parameter \\#1 \\$callback of function spl_autoload_register expects \\(callable\\(string\\)\\: void\\)\\|null, array\\{\\$this\\(CodeIgniter\\\\Autoloader\\\\Autoloader\\), 'loadClass'\\} given\\.$#"
+			message: "#^Parameter \\#1 \\$autoload_function of function spl_autoload_register expects callable\\(string\\)\\: void, array\\{\\$this\\(CodeIgniter\\\\Autoloader\\\\Autoloader\\), 'loadClass'\\} given\\.$#"
```

Comment:
This change should be reverted. PHPStan has different parameter names for the function based on 7.4 and 8.0, and PHPStan does not want to normalize them either. So I opted to run PHPStan on PHP 8.0+ so that the updated parameter name will be used.

Reply:
I didn't know it. It should be added in https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/pull_request.md#static-analysis-on-php-code

Reply:
Ok, that makes sense. Was wondering why it was failing now that it got pushed up here. Then should we set the ignore unused error to false? The main reason I was addressing these is because using 7.4 locally it was blocking me from committing any changes.

Reply:
I think phpstan has a parameter option to specify the php version it should "run".

Reply:
phpVersion

default: null (current PHP version is used)

If you want to analyse a codebase as if it was written for a different PHP version than you’re currently running, change the phpVersion parameter:

parameters:
    phpVersion: 70400 # PHP 7.4

Reply:
We should set it to 80000 to match our CICD

Reply:
I reverted the ignoreErrors and added phpVersion.

Reply:
Okay, all green!